src/java.base/share/classes/sun/security/provider/certpath/OCSPNonceExtension.java

```diff
@@ -135,13 +138,15 @@ public OCSPNonceExtension(boolean isCritical, byte[] incomingNonce)
         this.critical = isCritical;
 
         Objects.requireNonNull(incomingNonce, "Nonce data must be non-null");
-        if (incomingNonce.length > 0) {
+        // RFC 8954, section 2.1: the length of the nonce MUST be at least 1 octet
+        // and can be up to 32 octets.
+        if (incomingNonce.length > 0 && incomingNonce.length <=32) {
```

Comment:
nit: space after the <= to be consistent with style elsewhere

Reply:
fixed.